dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/RelationshipTypeControllerTest.java

```diff
@@ -0,0 +1,109 @@
+ */
+
+class RelationshipTypeControllerTest extends DhisControllerIntegrationTest
+{
+
+    private String program;
+
+    private String attrA;
+
+    private String attrB;
+
+    private String attrC;
+
+    private String trackedEntityType;
+
+    @BeforeEach
+    void setUp()
+    {
+        trackedEntityType = assertStatus( HttpStatus.CREATED,
+            POST( "/trackedEntityTypes/",
+                "{'name':'person', 'shortName':'person', 'description':'person', 'allowAuditLog':false }" ) );
+
+        attrA = assertStatus( HttpStatus.CREATED,
+            POST( "/trackedEntityAttributes/",
+                "{'name':'attrA', 'shortName':'attrA', 'valueType':'TEXT', 'aggregationType':'NONE'}" ) );
+
+        attrB = assertStatus( HttpStatus.CREATED,
+            POST( "/trackedEntityAttributes/",
+                "{'name':'attrB', 'shortName':'attrB', 'valueType':'TEXT', 'aggregationType':'NONE'}" ) );
+
+        attrC = assertStatus( HttpStatus.CREATED,
+            POST( "/trackedEntityAttributes/",
+                "{'name':'attrC', 'shortName':'attrC', 'valueType':'TEXT', 'aggregationType':'NONE'}" ) );
+
+        program = assertStatus( HttpStatus.CREATED, POST( "/programs/",
+            "{'name':'test program', 'id':'VoZMWi7rBgj', 'shortName':'test program','programType':'WITH_REGISTRATION', 'trackedEntityType': {"
+                +
+                "'id': '" + trackedEntityType + "'}," +
+                " 'programTrackedEntityAttributes' :  [{ 'trackedEntityAttribute' :{'id': '" + attrA
+                + "' }}, { 'trackedEntityAttribute' :{'id': '" + attrB + "' }}, { 'trackedEntityAttribute' :{'id': '"
+                + attrC + "' }}] }" ) );
+
+    }
+
+    @Test
+    void testPostingRelationshipTypes()
+    {
+        String relationshipTypeId = assertStatus( HttpStatus.CREATED,
+            POST( "/relationshipTypes/",
+                "{'code': 'test-rel','description': 'test-rel','fromToName': 'A to B','toConstraint': { 'relationshipEntity': "
+                    +
+                    "'TRACKED_ENTITY_INSTANCE','trackedEntityType': {'id': '" + trackedEntityType
+                    + "'}, 'program': {'id': '" + program + "'}, 'trackerDataView': {" +
+                    "'attributes': ['" + attrA + "','" + attrA + "', '" + attrB + "','" + attrC + "']} }, " +
+                    "'fromConstraint': { 'relationshipEntity':" +
+                    "'TRACKED_ENTITY_INSTANCE' , 'trackedEntityType': {'id': '" + trackedEntityType
+                    + "'}, 'program': {'id': '" + program + "' }},'name': 'test-rel'}" ) );
+
+        JsonList<JsonString> attributes = GET(
+            "/relationshipTypes/" + relationshipTypeId + "?fields=toConstraint[trackerDataView[attributes]]" ).content()
+                .getList( "toConstraint.trackerDataView.attributes", JsonString.class );
+
+        assertEquals( 3, attributes.size() );
+        assertEquals( List.of( attrA, attrB, attrC ), attributes.toList( JsonString::string ) );
```

Comment:
asserting on size is not a good practice. In fact, it is something you are doing in the next assertion